src/gallery/slider.js

```diff
@@ -0,0 +1,248 @@
+					mediaInfo => {
+						item.insertAdjacentHTML(
+							'beforeend',
+							renderImageInfo( mediaInfo, gallery[ index ], lang
+							) )
+					} )
+			}
+		}
+	},
+
+	renderNext = ( offset = 1 ) => {
+		const slider = parentContainer.querySelector( `.${prefixClassname}` ),
+			items = slider.querySelectorAll( `.${prefixClassname}-item` ),
+			nextButton = slider.querySelector( '.next' ),
+			previousButton = slider.querySelector( '.previous' ),
+			next = current + offset,
+			item = items[ next ]
+
+		if ( item ) {
+			current += offset
+			nextButton.style.opacity = current === items.length - 1 ? '0.5' : '1'
+			previousButton.style.opacity = current === 0 ? '0.5' : '1'
+
+			// render image attribution element - current, next, previous
+			showImageAndInfo( current )
+			showImageAndInfo( current + 1 )
+			showImageAndInfo( current - 1 )
+		}
+
+		slider.style[ dir === 'ltr' ? 'marginLeft' : 'marginRight' ] = -clientWidth * current + 'px'
+	},
+
+	renderPrevious = () => {
+		renderNext( -1 )
+	},
+
+	applyGestureEvent = () => {
+		let temp = {
+			screenX: null,
+			originalMarginLeft: null,
+			currentMarginLeft: null,
+			originalTransition: null
+		}
+
+		const container = parentContainer.querySelector( `.${prefixClassname}` ),
+			marginLR = dir === 'ltr' ? 'marginLeft' : 'marginRight'
+
+		container.addEventListener( 'touchstart', e => {
+			temp.screenX = e.touches[ 0 ].clientX
+			temp.originalMarginLeft =
+                +window.getComputedStyle( container )[ marginLR ].slice( 0, -2 )
+			temp.currentMarginLeft =
+                +window.getComputedStyle( container )[ marginLR ].slice( 0, -2 )
+			temp.originalTransition = window.getComputedStyle( container ).transition
+			container.style.transition = 'unset'
+		} )
+		container.addEventListener( 'touchmove', e => {
+			const clientX = e.touches[ 0 ].clientX,
+				offset = clientX - temp.screenX
+			temp.currentMarginLeft = temp.originalMarginLeft + offset * ( dir === 'ltr' ? 1 : -1 )
+			container.style[ marginLR ] = temp.currentMarginLeft + 'px'
+			e.preventDefault()
+		} )
+		container.addEventListener( 'touchend', () => {
+			container.style.transition = temp.originalTransition
+			const diff = temp.originalMarginLeft - temp.currentMarginLeft
+
+			if ( Math.abs( diff / clientWidth ) > 0.4 ) {
+				renderNext( diff > 0 ? 1 : -1 )
+			} else {
+				container.style[ marginLR ] = temp.originalMarginLeft + 'px'
+			}
+		} )
+	},
+
+	onShowFn = () => {
+		const sliderContainer = parentContainer.querySelector( `.${prefixClassname}` ),
+			items = sliderContainer.querySelectorAll( `.${prefixClassname}-item` ),
+			nextButton = sliderContainer.querySelector( '.next' ),
+			previousButton = sliderContainer.querySelector( '.previous' )
+
+		renderNext( 0 )
+		applyGestureEvent()
+
+		if ( items.length === 1 ) {
+			previousButton.style.visibility = 'hidden'
+			nextButton.style.visibility = 'hidden'
+		} else {
+			nextButton.addEventListener( 'click', () => {
```

Comment:
We can apply this to `renderPrevious` but not `renderNext` as it requires first parameter; otherwise changing the method parameters could work but i think it's fine in this case.